Flaky test: prometheusreceiver: TestStartTimeMetricRegex failing

**Describe the bug**
`TestStartTimeMetricRegex` is failing, (it was rerun since, will add a link to a failed workflow run when it happens again)

```
make[2]: Entering directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/prometheusreceiver'
go test -race -timeout 300s --tags="" ./...
--- FAIL: TestStartTimeMetricRegex (12.72s)
    metrics_receiver_test.go:1379: 
        	Error Trace:	metrics_receiver_test.go:1379
        	            				metrics_receiver_test.go:1564
        	            				metrics_receiver_test.go:1520
        	Error:      	Not equal: 
        	            	expected: time.Date(1970, time.January, 1, 0, 6, 40, 800000000, time.UTC)
        	            	actual  : time.Date(1970, time.January, 1, 0, 0, 0, 0, time.UTC)
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 (time.Time) {
        	            	- wall: (uint64) 800000000,
        	            	- ext: (int64) 62135597200,
        	            	+ wall: (uint64) 0,
        	            	+ ext: (int64) 62135596800,
        	            	  loc: (*time.Location)(<nil>)
        	Test:       	TestStartTimeMetricRegex
        	Messages:   	label_values:{value:"400" has_value:true} label_values:{value:"post" has_value:true} points:{timestamp:{seconds:1634220661 nanos:999000000} double_value:nan}
    metrics_receiver_test.go:1379: 
        	Error Trace:	metrics_receiver_test.go:1379
        	            				metrics_receiver_test.go:1564
        	            				metrics_receiver_test.go:1520
        	Error:      	Not equal: 
        	            	expected: time.Date(1970, time.January, 1, 0, 6, 40, 800000000, time.UTC)
        	            	actual  : time.Date(1970, time.January, 1, 0, 0, 0, 0, time.UTC)
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 (time.Time) {
        	            	- wall: (uint64) 800000000,
        	            	- ext: (int64) 62135597200,
        	            	+ wall: (uint64) 0,
        	            	+ ext: (int64) 62135596800,
        	            	  loc: (*time.Location)(<nil>)
        	Test:       	TestStartTimeMetricRegex
        	Messages:   	label_values:{value:"200" has_value:true} label_values:{value:"post" has_value:true} points:{timestamp:{seconds:1634220661 nanos:999000000} double_value:nan}
    metrics_receiver_test.go:1379: 
        	Error Trace:	metrics_receiver_test.go:1379
        	            				metrics_receiver_test.go:1564
        	            				metrics_receiver_test.go:1520
        	Error:      	Not equal: 
        	            	expected: time.Date(1970, time.January, 1, 0, 6, 40, 800000000, time.UTC)
        	            	actual  : time.Date(1970, time.January, 1, 0, 0, 0, 0, time.UTC)
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 (time.Time) {
        	            	- wall: (uint64) 800000000,
        	            	- ext: (int64) 62135597200,
        	            	+ wall: (uint64) 0,
        	            	+ ext: (int64) 62135596800,
        	            	  loc: (*time.Location)(<nil>)
        	Test:       	TestStartTimeMetricRegex
        	Messages:   	points:{timestamp:{seconds:1634220661 nanos:999000000} distribution_value:{count:-9223372036854775808 buckets:{count:-9223372036854775808}}}
    metrics_receiver_test.go:1379: 
        	Error Trace:	metrics_receiver_test.go:1379
        	            				metrics_receiver_test.go:1564
        	            				metrics_receiver_test.go:1520
        	Error:      	Not equal: 
        	            	expected: time.Date(1970, time.January, 1, 0, 6, 40, 800000000, time.UTC)
        	            	actual  : time.Date(1970, time.January, 1, 0, 0, 0, 0, time.UTC)
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 (time.Time) {
        	            	- wall: (uint64) 800000000,
        	            	- ext: (int64) 62135597200,
        	            	+ wall: (uint64) 0,
        	            	+ ext: (int64) 62135596800,
        	            	  loc: (*time.Location)(<nil>)
        	Test:       	TestStartTimeMetricRegex
        	Messages:   	points:{timestamp:{seconds:1634220661 nanos:999000000} summary_value:{count:{value:-9223372036854775808} sum:{} snapshot:{percentile_values:{percentile:1 value:nan}}}}
    metrics_receiver_test.go:1384: 
        	Error Trace:	metrics_receiver_test.go:1384
        	            				metrics_receiver_test.go:1564
        	            				metrics_receiver_test.go:1520
        	Error:      	Not equal: 
        	            	expected: 11
        	            	actual  : 22
        	Test:       	TestStartTimeMetricRegex
FAIL
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver	38.324s
```

**Steps to reproduce**
N/A

**What did you expect to see?**
No test failure

**What did you see instead?**
Test failure

**What version did you use?**
Version: https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/52f5f2b111dfd4d0d892c52fcc93c50382fac2c5 (from this PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/5694)

**What config did you use?**
N/A

**Environment**
N/A -> CI run

--- a/receiver/prometheusreceiver/metrics_receiver_test.go
+++ b/receiver/prometheusreceiver/metrics_receiver_test.go
@@ -1364,18 +1364,19 @@ var startTimeMetricPageStartTimestamp = &timestamppb.Timestamp{Seconds: 400, Nan
 const numStartTimeMetricPageTimeseries = 11
 
 func verifyStartTimeMetricPage(t *testing.T, _ *testData, mds []*agentmetricspb.ExportMetricsServiceRequest) {
+	if len(mds) < 1 {
+		t.Fatal("At least one metric request should be present")
+	}
 	numTimeseries := 0
-	for _, cmd := range mds {
-		for _, metric := range cmd.Metrics {
-			timestamp := startTimeMetricPageStartTimestamp
-			switch metric.GetMetricDescriptor().GetType() {
-			case metricspb.MetricDescriptor_GAUGE_DOUBLE, metricspb.MetricDescriptor_GAUGE_DISTRIBUTION:
-				timestamp = nil
-			}
-			for _, ts := range metric.GetTimeseries() {
-				assert.Equal(t, timestamp.AsTime(), ts.GetStartTimestamp().AsTime(), ts.String())
-				numTimeseries++
-			}
+	for _, metric := range mds[0].Metrics {
+		timestamp := startTimeMetricPageStartTimestamp
+		switch metric.GetMetricDescriptor().GetType() {
+		case metricspb.MetricDescriptor_GAUGE_DOUBLE, metricspb.MetricDescriptor_GAUGE_DISTRIBUTION:
+			timestamp = nil
+		}
+		for _, ts := range metric.GetTimeseries() {
+			assert.Equal(t, timestamp.AsTime(), ts.GetStartTimestamp().AsTime(), ts.String())
+			numTimeseries++
 		}
 	}
 	assert.Equal(t, numStartTimeMetricPageTimeseries, numTimeseries)
@@ -1521,7 +1522,7 @@ func TestStartTimeMetricRegex(t *testing.T) {
 func testEndToEndRegex(t *testing.T, targets []*testData, useStartTimeMetric bool, startTimeMetricRegex string) {
 	// 1. setup mock server
 	mp, cfg, err := setupMockPrometheus(targets...)
-	require.Nilf(t, err, "Failed to create Promtheus config: %v", err)
+	require.Nilf(t, err, "Failed to create Prometheus config: %v", err)
 	defer mp.Close()
 
 	cms := new(consumertest.MetricsSink)

--- a/receiver/prometheusreceiver/metrics_receiver_test.go
+++ b/receiver/prometheusreceiver/metrics_receiver_test.go
@@ -1364,9 +1364,7 @@ var startTimeMetricPageStartTimestamp = &timestamppb.Timestamp{Seconds: 400, Nan
 const numStartTimeMetricPageTimeseries = 11
 
 func verifyStartTimeMetricPage(t *testing.T, _ *testData, mds []*agentmetricspb.ExportMetricsServiceRequest) {
-	if len(mds) < 1 {
-		t.Fatal("At least one metric request should be present")
-	}
+	require.Greater(t, len(mds), 0, "At least one metric request should be present")
 	numTimeseries := 0
 	for _, metric := range mds[0].Metrics {
 		timestamp := startTimeMetricPageStartTimestamp

